keps/sig-scheduling/2891-simplified-config/README.md

```diff
@@ -0,0 +1,861 @@
+        {Name: names.GCEPDLimits},
+        {Name: names.NodeVolumeLimits},
+        {Name: names.AzureDiskLimits},
+        {Name: names.VolumeZone},
+        {Name: names.DefaultPreemption},
+        {Name: names.NodeResourcesBalancedAllocation, Weight: pointer.Int32Ptr(1)},
+        {Name: names.ImageLocality, Weight: pointer.Int32Ptr(1)},
+        {Name: names.DefaultBinder},
+      },
+    },
+  }
+}
+```
+(Note: Specific ordering for individual extension points will be based on the order 
+plugins are listed in any config)
+
+In this configuration, internal defaults for weight (which is only relevant for `Score`) 
+can still be propagated when the plugins are initialized after calling their `Register()` function 
+by modifying the [current weight registration code](https://github.com/kubernetes/kubernetes/blob/dc079acc2be603fbc2065e2748c79bb5e2c3453d/pkg/scheduler/framework/runtime/framework.go#L292-L306) 
+to check `MultiPoint.Enabled`:
+
+```go
+var totalPriority int64
+for _, e := range profile.Plugins.MultiPoint.Enabled {
+  // a weight of zero is not permitted, plugins can be disabled explicitly
+  // when configured.
+  f.scorePluginWeight[e.Name] = int(e.Weight)
+  if f.scorePluginWeight[e.Name] == 0 {
+    f.scorePluginWeight[e.Name] = 1
+  }
+
+  // Checks totalPriority against MaxTotalScore to avoid overflow
+  if int64(f.scorePluginWeight[e.Name])*framework.MaxNodeScore > framework.MaxTotalScore-totalPriority {
+    return nil, fmt.Errorf("total score of Score plugins could overflow")
+  }
+  totalPriority += int64(f.scorePluginWeight[e.Name]) * framework.MaxNodeScore
+}
+```
+
+Below are the semantics of enabling/disabling a plugin in the config, compared to current 
+regular extension points:
+
+|   |   |enabled: xyz|disabled: xyz|enabled: *|disabled: *|
+|---|---|------------|-------------|----------|-----------|
+|in-tree|regular|reorder plugins/set weight|disable plugin xyz|no-op|disable all plugins of the point|\
+|   |multi-point|enable plugin xyz for all extension points|disable plugin xyz for all extension points|no-op|disable all in-tree plugins for all extension points|
+|out-of-tree|regular|enable plugin xyz|no-op|no-op|no-op|
+|   |multi-point|enable plugin xyz for all extension points|no-op|no-op|no-op|
+
+Individual plugins may be disabled using `PluginArgs` to pass a config option for which extension 
+points the factory functions should register.
+
+#### Option 2
+
+Modify [`f.pluginsNeeded`](https://github.com/kubernetes/kubernetes/blob/dc079acc2be603fbc2065e2748c79bb5e2c3453d/pkg/scheduler/framework/runtime/framework.go#L1164) 
+to additionally parse MultiPoint plugins so the function 
+properly builds the map of `pluginName->New function`. Otherwise, we will not have a 
+reference to any plugins that are only enabled through MultiPoint later when we 
+try to unroll them to their individual extension points.
+
+```go
+func (f *frameworkImpl) pluginsNeeded(plugins *config.Plugins) map[string]config.Plugin {
```

Comment:
Leave code for code reviews :)

In most of these cases, it's enough to point to the functions were you need to inject new code and describe the intention of the change.

Reply:
I still stand that we don't need the code in the KEP

Reply:
+1 removed